pkgs/by-name/gm/gmt/package.nix

```diff
@@ -0,0 +1,104 @@
+  darwin,
+  fftwSinglePrec,
+  netcdf,
+  pcre,
+  gdal,
+  blas,
+  lapack,
+  glibc,
+  ghostscript,
+  dcw-gmt,
+  gshhg-gmt,
+  libxml2,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gmt";
+  version = "6.5.0";
+  src = fetchFromGitHub {
+    owner = "GenericMappingTools";
+    repo = "gmt";
+    rev = "refs/tags/${finalAttrs.version}";
+    hash = "sha256-KKIYhljCtk9t9CuvTLsSGvUkUwazWTm9ymBB3wLwSoI=";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  env = {
+    NIX_LDFLAGS = "-lxml2 -L${lib.getLib (libxml2.override { enableHttp = true; })}/lib";
+  };
+
+  buildInputs =
+    [
+      curl
+      gdal
+      netcdf
+      pcre
+      dcw-gmt
+      gshhg-gmt
+    ]
+    ++ (
+      if stdenv.isDarwin then
+        with darwin.apple_sdk.frameworks;
+        [
+          Accelerate
+          CoreGraphics
+          CoreVideo
+        ]
+      else
+        [
+          glibc
+          fftwSinglePrec
+          blas
+          lapack
+        ]
+    );
+
+  propagatedBuildInputs = [ ghostscript ];
+
+  cmakeFlags =
+    [
+      (lib.cmakeFeature "GMT_DOCDIR" "share/doc/gmt")
+      (lib.cmakeFeature "GMT_MANDIR" "share/man")
+      (lib.cmakeFeature "GMT_LIBDIR" "lib")
+      (lib.cmakeBool "COPY_GSHHG" false)
+      (lib.cmakeFeature "GSHHG_ROOT" "${gshhg-gmt.out}/share/gshhg-gmt")
+      (lib.cmakeBool "COPY_DCW" false)
+      (lib.cmakeFeature "DCW_ROOT" "${dcw-gmt.out}/share/dcw-gmt")
+      (lib.cmakeFeature "GDAL_ROOT" "${gdal.out}")
+      (lib.cmakeFeature "NETCDF_ROOT" "${netcdf.out}")
+      (lib.cmakeFeature "PCRE_ROOT" "${pcre.out}")
+      (lib.cmakeBool "GMT_INSTALL_TRADITIONAL_FOLDERNAMES" false)
+      (lib.cmakeBool "GMT_ENABLE_OPENMP" true)
+      (lib.cmakeBool "GMT_INSTALL_MODULE_LINKS" false)
+      (lib.cmakeFeature "LICENSE_RESTRICTED" "LGPL")
+    ]
+    ++ (lib.optionals (!stdenv.isDarwin) [
+      (lib.cmakeFeature "FFTW3_ROOT" "${fftwSinglePrec.dev}")
+      (lib.cmakeFeature "LAPACK_LIBRARY" "${lib.getLib lapack}/lib/liblapack.so")
+      (lib.cmakeFeature "BLAS_LIBRARY" "${lib.getLib blas}/lib/libblas.so")
+    ]);
+
+  meta = with lib; {
+    homepage = "https://www.generic-mapping-tools.org";
+    description = "Tools for manipulating geographic and cartesian data sets";
+    longDescription = ''
+      GMT is an open-source collection of command-line tools for manipulating
+      geographic and Cartesian data sets (including filtering, trend fitting,
+      gridding, projecting, etc.) and producing high-quality illustrations
+      ranging from simple x–y plots via contour maps to artificially illuminated
+      surfaces and 3D perspective views. It supports many map projections and
+      transformations and includes supporting data such as coastlines, rivers,
+      and political boundaries and optionally country polygons.
+    '';
+    platforms = lib.platforms.unix;
+    license = licenses.lgpl3Plus;
+    maintainers = with maintainers; [ tviti ];
+    broken = stdenv.isDarwin && stdenv.isAarch64;
```

Comment:
```suggestion
```

Reply:
Should be fine now after adding `-Wno-implicit-function-declaration`.

Reply:
That resolved one of the compile issues👍 yet the following still remains:

```
  /tmp/nix-build-gmt-6.5.0.drv-1/source/src/gmt_vector.c:1374:20: error: unknown type name '__LAPACK_int'
              (const __LAPACK_int * _Nonnull)&na, (const __LAPACK_int * _Nonnull)&nb, (const __LAPACK_int * _Nonnull)&nc,
```